nengo_gui/components/spike_grid.py

```diff
@@ -0,0 +1,73 @@
+import nengo
+import numpy as np
+import struct
+
+from nengo_gui.components.component import Component
+
+class SpikeGrid(Component):
+    def __init__(self, obj, n_neurons=None):
+        super(SpikeGrid, self).__init__()
+        self.obj = obj
+        self.data = []
+        self.max_neurons = self.obj.neurons.size_out
+        if n_neurons is None:
+            n_neurons = self.max_neurons
+        self.n_neurons = n_neurons
+        self.pixels_x = np.ceil(np.sqrt(self.n_neurons))
+        self.pixels_y = np.ceil(float(self.n_neurons) / self.pixels_x)
+        self.n_pixels = self.pixels_x * self.pixels_y
+        self.struct = struct.Struct('<f%dB' % (self.n_pixels))
+        self.max_value = 1.0
+
+    def attach(self, page, config, uid):
+        super(SpikeGrid, self).attach(page, config, uid)
+        self.label = page.get_label(self.obj)
+
+    def add_nengo_objects(self, page):
+        with page.model:
+            self.node = nengo.Node(self.gather_data,
+                                   size_in=self.obj.neurons.size_out)
+            self.conn = nengo.Connection(self.obj.neurons,
+                                         self.node, synapse=0.01)
+
+    def remove_nengo_objects(self, page):
+        page.model.connections.remove(self.conn)
+        page.model.nodes.remove(self.node)
+
+    def gather_data(self, t, x):
+        self.max_value = max(self.max_value, np.max(x))
+
+        # TODO: pass only spiking neurons, using subclass of Nengo.Image?
```

Comment:
Ah, that comment's a bit outdated anyway, as we don't want to just send spike data when we're doing the synapse filter in Nengo.  I removed it.